calico_containers/pycalico/block.py

```diff
@@ -214,10 +237,12 @@ def auto_assign(self, num, handle_id, attributes, host,
                 ips.append(ip)
         return ips
 
-    def assign(self, address, handle_id, attributes):
+    def assign(self, address, handle_id, attributes, host):
```

Comment:
i think that makes for a yuckier interface since it requires the user to check that parameter before deciding whether to pass in the host.  

Further afield we might need to pass in a bunch of different affinity values (rack, host etc ... ), but since this interface is used internally by the IPAM client, I don't think it is worth designing that at this point.